pallene/parser.lua

```diff
@@ -88,18 +116,24 @@ function Parser:Program()
     while not self:peek("EOF") do
         table.insert(tls, self:Toplevel())
     end
-    return tls
+    return {
+        tls = tls,
+        regions = self.regions
+    }
```

Comment:
What do you think about creating a new kind of AST node, ast.Program.Program? It would be nice to have all the field names listed in ast.lua.